tests/test_BT_piecewise.py

```diff
@@ -0,0 +1,560 @@
+from pint.models import get_model
+import pint.toa
+import numpy as np
+import pint.fitter
+import astropy.units as u
+from pint import ls
+from copy import deepcopy
+import pint.residuals
+from astropy.time import Time
+import pint.models.stand_alone_psr_binaries.BT_piecewise as BTpiecewise
+import matplotlib.pyplot as plt
+import unittest
+from io import StringIO
+from pylab import *
+import pytest
+import pint.residuals
+import pint.simulation
+
+
+@pytest.fixture
+def build_model():
+    # builds a J1023+0038-like model with no pieces
+    par_base = """
+    PSR                             1023+0038
+    TRACK                                  -3
+    EPHEM                               DE421
+    CLOCK                        TT(BIPM2019)
+    START              54906.7327557001113774
+    FINISH             56443.7950227385911112
+    DILATEFREQ                              N
+    DMDATA                                0.0
+    NTOA                               4970.0
+    CHI2                   14234223.733031187
+    RAJ                     10:23:47.68719801
+    DECJ                     0:38:40.84551000
+    PMRA                                  0.0
+    PMDEC                                 0.0
+    PX                                    0.0
+    POSEPOCH           54995.0000000000000000
+    F0                   592.4214681620181649 0 
+    F1              -2.4181868552365951721e-15 0
+    PEPOCH             55352.6375960000000001
+    PLANET_SHAPIRO                          N
+    DM                  14.720549454038943568 1 
+    DM1                                   0.0
```

Comment:
Can you omit some of the less important bits, like DM1 and PMRA? Keep this a little shorter and easier to glance over? Which bits are actually needed?